pkg/health/server/server.go

```diff
@@ -264,14 +283,56 @@ func (s *Server) collectNodeConnectivityMetrics() {
 				location, metrics.LabelPeerNode, metrics.LabelTrafficICMP, metrics.LabelAddressTypeSecondary)
 		}
 	}
+
+	// Aggregated health statuses for endpoint connectivity
+	metrics.NodeHealthConnectivityStatus.WithLabelValues(
+		localClusterName, localNodeName, metrics.LabelPeerEndpoint, metrics.LabelReachable).
+		Set(float64(endpointStatuses[healthClientPkg.ConnStatusReachable]))
+
+	metrics.NodeHealthConnectivityStatus.WithLabelValues(
+		localClusterName, localNodeName, metrics.LabelPeerEndpoint, metrics.LabelUnreachable).
+		Set(float64(endpointStatuses[healthClientPkg.ConnStatusUnreachable]))
+
+	metrics.NodeHealthConnectivityStatus.WithLabelValues(
+		localClusterName, localNodeName, metrics.LabelPeerEndpoint, metrics.LabelUnknown).
+		Set(float64(endpointStatuses[healthClientPkg.ConnStatusUnknown]))
+
+	// Aggregated health statuses for node connectivity
+	metrics.NodeHealthConnectivityStatus.WithLabelValues(
+		localClusterName, localNodeName, metrics.LabelPeerNode, metrics.LabelReachable).
+		Set(float64(nodeStatuses[healthClientPkg.ConnStatusReachable]))
+
+	metrics.NodeHealthConnectivityStatus.WithLabelValues(
+		localClusterName, localNodeName, metrics.LabelPeerNode, metrics.LabelUnreachable).
+		Set(float64(nodeStatuses[healthClientPkg.ConnStatusUnreachable]))
+
+	metrics.NodeHealthConnectivityStatus.WithLabelValues(
+		localClusterName, localNodeName, metrics.LabelPeerNode, metrics.LabelUnknown).
+		Set(float64(nodeStatuses[healthClientPkg.ConnStatusUnknown]))
 }
 
 func collectConnectivityMetric(status *healthModels.ConnectivityStatus, labels ...string) {
-	var metricValue float64 = -1
 	if status != nil {
-		metricValue = float64(status.Latency) / float64(time.Second)
+		// node_health_connectivity_latency_seconds will replace the now deprecated
+		// node_connectivity_latency_seconds and copies a subset of the labels for use
+		// when observing metrics
+		if len(labels) < 7 {
+			log.Warn("node_connectivity_latency_seconds metric is missing labels")
+			return
+		}
+		healthLabels := make([]string, 5)
+		copy(healthLabels, labels[0:2])
+		copy(healthLabels[2:], labels[6:])
+		if status.Status == "" {
+			metricValue := float64(status.Latency) / float64(time.Second)
+			metrics.NodeConnectivityLatency.WithLabelValues(labels...).Set(metricValue)
+			metrics.NodeHealthConnectivityLatency.WithLabelValues(healthLabels...).Observe(metricValue)
+		} else {
+			metrics.NodeHealthConnectivityLatency.WithLabelValues(healthLabels...).Observe(probeTimeout)
```

Comment:
Seems like you changed the behavior of the old metric:
before, it was always reporting some value for metric `NodeConnectivityLatency`.
Now, if the probe times out `(status.Status != "")`, it won't be updated. I would suggest splitting this part of the code into two parts:
- leave the previous code as it was
- new part with new metrics
Something like this (I've also remove indentation of ifs here):
```
// Old metric
var metricValue float64 = -1
if status != nil {
  metricValue = float64(status.Latency) / float64(time.Second)
}
metrics.NodeConnectivityLatency.WithLabelValues(labels...).Set(metricValue)
// New metric
if status == nil {
  // We don't report unperformed probes
  return
}
if len(labels) < 7 {
  ...
}
```
It will be much simpler to read/safer and removed once deprecation is completed :)

Reply:
Thank you, sorry I seem to have missed this case. Makes sense to be about making it easier to deprecate/read and safer, I will keep it in mind :) Thank you Marcel!